lib/decryptor.rb

```diff
@@ -0,0 +1,33 @@
+require_relative './cryptor'
+require_relative './key'
+require_relative './offset'
+
+class Decryptor < Cryptor
+
+  attr_reader :decrypt_key, :decrypt_offset, :decrypt_shift
+
+  def initialize(message, key, date = '') 
+    super(message, key, date) 
+    @decrypt_key = Key.new(key)
+    @decrypt_offset = (date == '' ? Offset.new : Offset.new(date))
+    @shift = [@decrypt_key.make_keys, @decrypt_offset.make_offsets].transpose.map{|a| a.sum} 
+  end
+
+  def unshifted_alphabet_positions
+    result = []
+    message_in_alphabet_positions.each_with_index do |position, index|
+      position.nil? ? result << position : result << (position - message_shifts[index]) % 27 
```

Comment:
In order to undo the work done during encryption, shifts are subtracted from the positions in the message. This method could also be made more general and moved to the `Cryptor` super class